src/main/java/xyz/nucleoid/plasmid/command/MapCommand.java

```diff
@@ -268,4 +318,34 @@ private static int commitRegion(CommandContext<ServerCommandSource> context) thr
             );
         };
     }
+
+    @FunctionalInterface
+    private interface MapExecutor {
+        int execute(CommandContext<ServerCommandSource> context, StagingMapTemplate map, BlockPos pos) throws CommandSyntaxException;
+
+        static Command<ServerCommandSource> make(MapExecutor executor) {
```

Comment:
It might be simpler to have this work as like `StagingMapManager getMap(CommandContext<ServerCommandSource>) throws CommandSyntaxException`?